modules/consensus/accept.go

```diff
@@ -55,6 +56,73 @@ func (cs *ConsensusSet) validateHeader(tx dbTx, b types.Block) error {
 	return cs.blockValidator.ValidateBlock(b, minTimestamp, parent.ChildTarget, parent.Height+1)
 }
 
+// checkTarget returns true if the header's ID meets the given target.
+func checkHeaderTarget(h types.BlockHeader, target types.Target) bool {
+	blockHash := h.ID()
+	return bytes.Compare(target[:], blockHash[:]) >= 0
+}
+
+// validateHeader does some early, low computation verification on the header
+// to determine if the block should be downloaded. Callers should not assume
+// that validation will happen in a particular order.
+func (cs *ConsensusSet) validateHeader(tx dbTx, h types.BlockHeader) error {
+	// See if the block is known already.
+	id := h.ID()
+	_, exists := cs.dosBlocks[id]
+	if exists {
+		return errDoSBlock
+	}
+
+	// Check if the block is already known.
+	blockMap := tx.Bucket(BlockMap)
+	if blockMap == nil {
+		return errNoBlockMap
+	}
+	if blockMap.Get(id[:]) != nil {
+		return modules.ErrBlockKnown
+	}
+
+	// Check for the parent.
+	parentID := h.ParentID
+	parentBytes := blockMap.Get(parentID[:])
+	if parentBytes == nil {
+		return errOrphan
+	}
+	var parent processedBlock
+	err := cs.marshaler.Unmarshal(parentBytes, &parent)
+	if err != nil {
+		return err
+	}
+
+	// Check that the target of the new block is sufficient.
+	if !checkHeaderTarget(h, parent.ChildTarget) {
+		return modules.ErrBlockUnsolved
+	}
+
+	// TODO: check if the block is a non extending block.
```

Comment:
you can determine that with just a header, though you don't want to trust it because you don't actually know if the data for the block exists. And you don't want to skip downloading the block if it's non-extending because you need to be able to get reorgs.

A check here doesn't make sense until we're doing headers-first downloads.